Documentation and test fixes

GiudGiud opened the pull request:
- [x] Need to rework commit history
- [x] Need to pass tests :)

Any opposition to removing the bibliography section in doco pages that do not cite anything?

After this PR and another to address #20522, and then https://github.com/idaholab/moose/pull/19578 as well, we'll have very few deprecation failures within the modules, so people will be able to use --error-deprecated on the combined test suite to find all new instances of deprecation

GiudGiud committed:
Add a check reset_ parameters for MultiApps
Closes #20540

Fixup the fix for reset parameter

GiudGiud committed:
Remove empty source files, which are only used to force compiling of headers

GiudGiud committed:
Miscellaneous fixes on VPP docs
Refs #18333

GiudGiud committed:
Fix documentation for two postprocessors
Refs #18333

GiudGiud committed:
More snippet fixes for pinsfv doc
Refs #16756

GiudGiud committed:
Add documentation to Table Output
refs #18333 #17566 #16248 #15968

GiudGiud committed:
More docs for ElementExtremeMatprop, add docs for OutputInterface
refs #18333 #17566 #16248 #15968

GiudGiud committed:
Misc comment fixes in code
Remove unused variable in TableOutput

GiudGiud committed:
Fixup moosedocs for numerous classes. Typo fixes. Move ParisLaw doc to right module

Fixup documentation fixes, refs #18333

Move ParisLaw correctly to XFEM

fixup! Fixup moosedocs for numerous classes. Typo fixes. Move ParisLaw doc to right module

Fixup docs for scalar tag aux

GiudGiud committed:
Fix doc files which refer to wrong source file for their parameters
refs #18333

GiudGiud committed:
Fix parameter description in FileOutput
Fix typo in GapConductanceConstraint
refs #18333

More work on file output docs

GiudGiud committed:
Remove unused parameter from MOOSE example 1, refs #20510

GiudGiud committed:
Fix some deprecated parameters in tensor mechanics module
Waiting for guidance on others, see #20522

GiudGiud committed:
Fix some deprecated parameters in contact module tests, refs #0 and linked to  #15283

GiudGiud committed:
Fix some deprecated parameters in heat conduction tests, refs #0

GiudGiud committed:
Fix some deprecated parameters in porous flow tests, refs #0

GiudGiud committed:
Fix some deprecated parameters in a phase field example, refs #0

GiudGiud committed:
Fix some deprecated parameters in a reactor module test, refs #0

GiudGiud committed:
Fix some deprecated parameters in ray tracing fv tests, refs #0

GiudGiud committed:
Add more instructions to INL HPC install, refs #20071

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/980779/) on eed7259 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20525/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/980802/) on eed7259 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>decd40</th>
      <th colspan=4>#20525 eed725</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.42%</td>
      <td>82.41%</td>
      <td>-0.00%</td>
      <td>83.33%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>73437</td>
      <td>73440</td>
      <td>+3</td>
      <td>5</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15669</td>
      <td>15670</td>
      <td>+1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20525/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20525/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20525/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 83.33% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud committed:
Add parameter groups to MultiApps, refs #20535

GiudGiud committed:
Fix parameter groups for fixed point solves, refs #20535

GiudGiud committed:
Fix up deprecation messages in PicardSolve
    refs #17479

GiudGiud committed:
Add parameter groups to FEProblemSolve, refs #20535

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/982181/) on 1288a92 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/20525/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/20525/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 2f149d0299277a6692b6976693d2669b94a272de`

GiudGiud committed:
Fix some deprecated parameters in stochastic tools test, refs #0
This loses the diffs on those reporters, since the reporters dont output like VPPs
BUT it's the stats of results, which is directly output as well and is checked. So the example in itself remains covered. The correct working of StatisticsReporter is also ensured by the test suite, not by these examples
lang format

cticenhour commented on framework/doc/content/source/variables/MooseVariableFE.md:
```diff
@@ -1 +1,11 @@
+
+!syntax description /Variables/MooseVariable
+
+!! MOOSE Documentation Stub: Remove this line when content is added.
```
```suggestion
!alert construction title=Undocumented
```

cticenhour commented on framework/doc/content/source/vectorpostprocessors/CSVReader.md:
```diff
@@ -6,8 +6,13 @@ converts each column into a VectorPostprocessor vector. This object uses the
 
 ## Example Input Syntax
 
+In this example, the `example.csv` file containings data for year/month/day is being read by
```
```suggestion
In this example, the `example.csv` file containing data for year/month/day is being read by
```

cticenhour commented on framework/doc/content/source/vectorpostprocessors/LineMaterialRealSampler.md:
```diff
@@ -7,8 +7,21 @@
+
+- rows are ordered by point sampled along the line
+
+- columns are ordered by alphabetical order of the properties sampled. The distance along the sampled line, and the x, y and z coordinates of the sampled points are added to the output as additional columns.
```
```suggestion
- columns are ordered by alphabetical order of the properties sampled. The distance along the sampled line and the x, y and z coordinates of the sampled points are added to the output as additional columns.
```

GiudGiud commented:
If any of the 8 people requested for review need a chill PR to taper off into the weekend... this is it

cticenhour commented:
Working on this now....almost done.

GiudGiud committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

GiudGiud commented on framework/doc/content/source/outputs/OutputWarehouse.md:
```diff
@@ -1 +1,3 @@
 # OutputWarehouse
+
+!! MOOSE Documentation Stub: Remove this line when content is added.
```
Sure lemme know if you want the rest of the stub. It's very temporary, I'll fill those two last classes soon

cticenhour replied:
Leaving the rest of it in should be fine!

GiudGiud commented:
@cticenhour thanks for the review! I merged all your suggestions in.

GiudGiud commented:
Refs #20540 #18333 #20510 #20522 #15283 #20071 #20535 #17478
